### PR TITLE
MODLD-566: Tests for Suppress Flags feature

### PR DESCRIPTION
**Pull request by AndreiBordak**
## Purpose
Cover a scenario with Suppress flag update for Inventory instance

## Approach
Testing through API

### TODOS and Open Questions
- [x] Scenario failing on JSON conversion on step ```call putInventoryInstance```: ```json string can not be null or empty```

### CODE REVIEW

File: mod-linked-data/src/main/resources/citation/mod-linked-data/features/suppress-flags/samples/create-work-request.json [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            "http://bibfra.me/vocab/marc/partNumber":[
+              "1"
+            ],
+            "http://bibfra.me/vocab/marc/partName":[
+              "part 1"
+            ]
+          }
+        }
+      ],
+      "http://bibfra.me/vocab/marc/governmentPublication":[
```
Comment (pkjacob) [inline mod-linked-data/src/main/resources/citation/mod-linked-data/features/suppress-flags/samples/create-work-request.json:22]: Minor - You can simplify this request by removing everything other than "http://bibfra.me/vocab/marc/title" from the request.

File: mod-linked-data/src/main/resources/citation/mod-linked-data/features/suppress-flags/suppress-flags.feature [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    * def getInventoryInstanceResponse = call getInventoryInstance { id: "#(inventoryInstanceId)" }
+    * def inventoryInstance = getInventoryInstanceResponse.response
+    * eval inventoryInstance['staffSuppress'] = true
+    * eval inventoryInstance['discoverySuppress'] = true
+    * call putInventoryInstance
+
+    # Search the Check Suppress Flags values
+    Given path 'search/linked-data/works'
+    And param query = query
+    And retry until response.content[0].instances[0].suppress.staff == true && response.content[0].instances[0].suppress.fromDiscovery == true
```
Comment (pkjacob) [inline mod-linked-data/src/main/resources/citation/mod-linked-data/features/suppress-flags/suppress-flags.feature:33]: validate staff suppress also here.

Comment (AndreiBordak) [reply mod-linked-data/src/main/resources/citation/mod-linked-data/features/suppress-flags/suppress-flags.feature:33]: could you please clarify what do you mean PK? this step will pass only if both suppress.staff and suppress.fromDiscovery are true

Comment (pkjacob) [reply mod-linked-data/src/main/resources/citation/mod-linked-data/features/suppress-flags/suppress-flags.feature:33]: oh... I missed that. You are right

File: mod-linked-data/src/main/resources/citation/mod-linked-data/features/suppress-flags/samples/create-work-request.json (hunk 1 above)
Comment (AndreiBordak) [reply mod-linked-data/src/main/resources/citation/mod-linked-data/features/suppress-flags/samples/create-work-request.json:22]: already tried to do it, but test got failed:
image
so, decided to keep it as is

Comment (pkjacob) [reply mod-linked-data/src/main/resources/citation/mod-linked-data/features/suppress-flags/samples/create-work-request.json:22]: Following requests worked for me

{
  "resource":{
    "http://bibfra.me/vocab/lite/Work":{
      "http://bibfra.me/vocab/marc/title":[
        {
          "http://bibfra.me/vocab/marc/Title":{
            "http://bibfra.me/vocab/bflc/nonSortNum":[
              "4"
            ],
            "http://bibfra.me/vocab/marc/mainTitle":[
              "suppress_flags"
            ],
            "http://bibfra.me/vocab/marc/partNumber":[
              "1"
            ],
            "http://bibfra.me/vocab/marc/partName":[
              "part 1"
            ]
          }
        }
      ],
      "http://bibfra.me/vocab/marc/content":[
        {
          "http://bibfra.me/vocab/marc/term":[
            "cartographic image"
          ],
          "http://bibfra.me/vocab/lite/link":[
            "http://id.loc.gov/vocabulary/contentTypes/cri"
          ]
        }
      ]
    }
  }
}

File: mod-linked-data/src/main/resources/citation/mod-linked-data/features/suppress-flags/suppress-flags.feature [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    When method GET
```
Comment (pkjacob) [inline mod-linked-data/src/main/resources/citation/mod-linked-data/features/suppress-flags/suppress-flags.feature:34]: When I executed this test in my local, it failed with the following error
```
17:02:01.825 retry condition satisfied
17:02:01.827 classpath:citation/mod-linked-data/features/suppress-flags/suppress-flags.feature:34
When method GET
js failed:
>>>>
01: response.content[0].instances[0].suppress.staff == true && response.content[0].instances[0].suppress.fromDiscovery == true
<<<<
org.graalvm.polyglot.PolyglotException: TypeError: Cannot read property "staff" from undefined
- <js>.:program(Unnamed:1)

classpath:citation/mod-linked-data/features/suppress-flags/suppress-flags.feature:34
```

Are you getting any failure?

Comment (AndreiBordak) [reply mod-linked-data/src/main/resources/citation/mod-linked-data/features/suppress-flags/suppress-flags.feature:34]: No @pkjacob, i'm not, all tests are green:
image
image

Comment (AndreiBordak) [reply mod-linked-data/src/main/resources/citation/mod-linked-data/features/suppress-flags/suppress-flags.feature:34]: it's also green with the changes in the work sample file mentioned above.

File: mod-linked-data/src/main/resources/citation/mod-linked-data/features/suppress-flags/samples/create-work-request.json (hunk 1 above)
Comment (AndreiBordak) [reply mod-linked-data/src/main/resources/citation/mod-linked-data/features/suppress-flags/samples/create-work-request.json:22]: thank you PK, have updated pr

**Review by askhat-abishev — APPROVED** [review]

**Review by pkjacob — APPROVED** [review]

**Pull request merged**